File: logisim/app/src/main/java/logisim/gui/prefs/PrefOption.java
/* Copyright (c) 2010, Carl Burch. License information is located in the
 * logisim.Main source code and at www.cburch.com/logisim/. */

package logisim.gui.prefs;

import javax.swing.JComboBox;

class PrefOption {
	private Object value;
	private String displayString;

	PrefOption(String value, String displayString) {
		this.value = value;
		this.displayString = displayString;
	}

	@Override
	public String toString() {
		return displayString;
	}

	public Object getValue() {
		return value;
	}

	static void setSelected(JComboBox<PrefOption> combo, Object value) {
		for (int i = combo.getItemCount() - 1; i >= 0; i--) {
			PrefOption opt = combo.getItemAt(i);
			if (opt.getValue().equals(value)) {
				combo.setSelectedItem(opt);
				return;
			}
		}
		combo.setSelectedItem(combo.getItemAt(0));
	}

}

File: LogisimPlus/LogisimPlus/logisim/gui/prefs/PrefOption.cs
// ====================================================================================================
// Produced by the Free Edition of Java to C# Converter.
// To produce customized conversions, purchase a Premium Edition license:
// https://www.tangiblesoftwaresolutions.com/product-details/java-to-csharp-converter.html
// ====================================================================================================

/* Copyright (c) 2010, Carl Burch. License information is located in the
 * logisim.Main source code and at www.cburch.com/logisim/. */

namespace logisim.gui.prefs
{

	internal class PrefOption
	{
		private object value;
		private string displayString;

		internal PrefOption(string value, string displayString)
		{
			this.value = value;
			this.displayString = displayString;
		}

		public override string ToString()
		{
			return displayString;
		}

		public virtual object Value
		{
			get
			{
				return value;
			}
		}

		internal static void setSelected(JComboBox<PrefOption> combo, object value)
		{
			for (int i = combo.getItemCount() - 1; i >= 0; i--)
			{
				PrefOption opt = combo.getItemAt(i);
				if (opt.Value.Equals(value))
				{
					combo.setSelectedItem(opt);
					return;
				}
			}
			combo.setSelectedItem(combo.getItemAt(0));
		}

	}

}
